pallets/msa/src/lib.rs

```diff
@@ -275,6 +319,36 @@ pub mod pallet {
 		ProofHasExpired,
 		/// The submitted proof expiration block is too far in the future
 		ProofNotYetValid,
+		/// Attempted to add a signature when the signature is already in the registry
+		BucketKeyExists,
+		/// Attempted to add a signature with a mortality too far in the future
+		MortalityTooHigh,
+		/// Attempted to retrieve a bucket outside the bounds of storage
+		NoSuchBucket,
+	}
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		phantom: PhantomData<T>,
+	}
```

Comment:
Not sure why we need this, but I guess it might be for testing?

Reply:
I'm not sure which "this" you're referring to - if you mean the phantom data, it's because we need to use `T::Blocknumber` when initializing MortalityBlockOf, and rust requires that you use the generic, and Substrate requires the GenesisConfig struct to be defined.

If you mean why we need to initialize `MortalityBlockOf`, It was a decision between using a genesis config to set the very first values, versus including the logic in the extrinsic.  IMO setting genesis is better rather than having logic that will literally only ever be hit once.